docs/en/notes/contribution_guide.md

```diff
@@ -1,31 +1,234 @@
+
+or:
+
+pip install -U pre-commit -i https://pypi.mirrors.ustc.edu.cn/simple
+```
+
+If the installation process is interrupted, you can repeatedly run `pre-commit run ...` to continue the installation.
+
+If the code does not conform to the code style specification, pre-commit will raise a warning and fixes some of the errors automatically.
+
+![](https://user-images.githubusercontent.com/57566630/202369176-67642454-0025-4023-a095-263529107aa3.png)
+
+### 3. Create a development branch
+
+After configuring the pre-commit, we should create a branch based on the dev branch to develop the new feature or fix the bug. The proposed branch name is `username/pr_name`.
+
+```Shell
+git checkout -b username/refactor_contributing_doc
+```
 
-We adopt [PEP8](https://www.python.org/dev/peps/pep-0008/) as the preferred code style.
+In subsequent development, if the dev branch of the local repository lags behind the dev branch of the official repository, you need to pull the upstream dev branch first and then rebase it to the local development branch.
 
-We use the following tools for linting and formatting:
+```Shell
+git checkout dev-1.x
+
+git pull upstream dev-1.x
+
+git checkout username/refactor_contributing_doc
+
+git rebase dev-1.x
+```
+
+When rebasing, if a conflict arises, you need to resolve the conflict manually, then execute the `git add` command, and then execute the `git rebase --continue` command until the rebase is complete.
+
+### 4. Commit the code and pass the unit test
+
+After the local development is done, we need to pass the unit tests locally and then commit the code.
+
+```shell
+# run unit test
+pytest tests/
+
+# commit the code
+git add .
+git commit -m "commit message"
+```
+
+### 5. Push the code to the remote repository
+
+After the local development is done, we need to push the code to the remote repository.
+
+```Shell
+git push origin username/refactor_contributing_doc
+```
+
+### 6. Create a Pull Request
+
+#### (1) Create a Pull Request on GitHub
+
+![](https://user-images.githubusercontent.com/13503330/223321382-e6068e18-1d91-4458-8328-b1c7c907b3b2.png)
+
+#### (2) Fill in the Pull Request template
+
+![](https://user-images.githubusercontent.com/57566630/167307569-a794b967-6e28-4eac-a942-00deb657815f.png)
+
+## Guidance
+
+### Unit Test
+
+We should also make sure the committed code will not decrease the coverage of unit test, we could run the following command to check the coverage of unit test:
+
+```Shell
+python -m coverage run -m pytest tests/
+python -m coverage html
+# check file in htmlcov/index.html
+```
+
+### Document rendering
+
+If the documents are modified/added, we should check the rendering result. We could install the dependencies and run the following command to render the documents and check the results:
+
+```Shell
+pip install -r requirements/docs.txt
+cd docs/en/
+# or docs/zh_cn
+make html
+# check file in ./docs/en/_build/html/index.html
+```
```

Comment:
I think the unit test belongs to 'advanced' part. We can add docs about how do write ut script and how to run it locally in the future.

Online documentation is not what community contributors need to concern about (different from the case of MMYOLO).